punchcut/src/source/glyf/deltas.rs

```diff
@@ -0,0 +1,289 @@
+use read_fonts::{
+    tables::glyf::{PointFlags, PointMarker},
+    tables::gvar::{GlyphDelta, Gvar},
+    types::{F26Dot6, F2Dot14, Fixed, GlyphId, Point},
+    ReadError,
+};
+
+/// Compute a set of deltas for the component offsets of a composite glyph.
+///
+/// Interpolation is meaningless for component offsets so this is a
+/// specialized function that skips the expensive bits.
+pub fn composite_glyph(
+    gvar: &Gvar,
+    glyph_id: GlyphId,
+    coords: &[F2Dot14],
+    deltas: &mut [Point<Fixed>],
+) -> Result<(), ReadError> {
+    for delta in deltas.iter_mut() {
+        *delta = Default::default();
+    }
+    let Ok(var_data) = gvar.glyph_variation_data(glyph_id) else {
+        // Empty variation data for a glyph is not an error.
+        return Ok(());
+    };
+    for tuple in var_data.tuples() {
+        let Some(scalar) = tuple.compute_scalar(coords) else {
+            continue;
```

Comment:
It would be nice if the documentation for compute_scalar clarified it will never return 0, you get None or Some(non-zero value)

Reply:
Done